### PR TITLE
fix(torch backend): reintroduced necessary manual dtype casting.

### PR DESCRIPTION
**Pull request by Madjid-CH**
Revert unnecessary changes https://github.com/unifyai/ivy/pull/23659

### CODE REVIEW

**Comment by Madjid-CH** [comment]
Hey @vedpatwardhan you only need to approve this PR since you already reviewed all the changes [here](https://github.com/unifyai/ivy/pull/24615), in the mean while we see what @MahmoudAshraf97 say about this.
Thanks😄

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Let's see what @MahmoudAshraf97 thinks, thanks @Madjid-CH 😄

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]
everything else LGTM

File: ivy/functional/backends/torch/experimental/statistical.py [hunk 1]
```diff
@@ -581,7 +581,7 @@ def cummax(
     out: Optional[torch.Tensor] = None,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
     if x.dtype in (torch.complex64, torch.complex128):
-        x = x.real.to(dtype=torch.float64)
+        x = x.real
```
[the remark this answers is not in the document]
Comment (MahmoudAshraf97) [reply ivy/functional/backends/torch/experimental/statistical.py:584]: I don't think these are similar situations, the problem here is that you discarded the imaginary part of the array without reason to do so

[the remark this answers is not in the document]
Comment (Madjid-CH) [reply ivy/functional/backends/torch/experimental/statistical.py:584]: did it was like that before I change it ?
`x = x.real.to(dtype=torch.float64)`

[the remark this answers is not in the document]
Comment (Madjid-CH) [reply ivy/functional/backends/torch/experimental/statistical.py:584]: It was discarding the imaginary part by creating a new array, now it discard it without a new array 😄

[the remark this answers is not in the document]
Comment (MahmoudAshraf97) [reply ivy/functional/backends/torch/experimental/statistical.py:584]: yeah it was wrong from the beginning, we should add proper support or not support at all

[the remark this answers is not in the document]
Comment (Madjid-CH) [reply ivy/functional/backends/torch/experimental/statistical.py:584]: understood I'll change that

**Pull request merged**